packages/main/test/pages/TabContainerDragAndDrop.html

```diff
@@ -0,0 +1,159 @@
+			<ui5-tab-separator></ui5-tab-separator>
+			<ui5-tab id="tabEight" movable design="Positive" text="Eight"></ui5-tab>
+			<ui5-tab id="tabNine" movable design="Negative" text="Nine"></ui5-tab>
+			<ui5-tab id="tabTen" movable design="Critical" text="Ten"></ui5-tab>
+			<ui5-tab id="tabEleven" movable text="Eleven"></ui5-tab>
+			<ui5-tab-separator></ui5-tab-separator>
+			<ui5-tab id="tabTwelve" movable text="Twelve"></ui5-tab>
+			<ui5-tab id="tabThirteen" movable text="Thirteen" selected></ui5-tab>
+			<ui5-tab id="tabFourteen" movable text="Fourteen"></ui5-tab>
+			<ui5-tab id="tabFifteen" movable text="Fifteen"></ui5-tab>
+			<ui5-tab id="tabSixteen" movable text="Sixteen"></ui5-tab>
+			<ui5-tab id="tabSeventeen" movable text="Seventeen (forbids nesting)"></ui5-tab>
+			<ui5-tab id="tabEighteen" movable text="Eighteen"></ui5-tab>
+			<ui5-tab-separator></ui5-tab-separator>
+			<ui5-tab id="tabNinteen" movable text="Nineteen"></ui5-tab>
+			<ui5-tab id="tabTwenty" movable text="Twenty"></ui5-tab>
+			<ui5-tab id="tabTwentyOne" movable text="Twenty One"></ui5-tab>
+			<ui5-tab id="tabTwentyTwo" movable text="Twenty Two"></ui5-tab>
+			<ui5-tab id="tabTwentyThree" movable text="Twenty Three"></ui5-tab>
+			<ui5-tab id="tabTwentyFour" movable text="Twenty Four"></ui5-tab>
+			<ui5-tab-separator></ui5-tab-separator>
+			<ui5-tab id="tabTwentyFive" movable text="Twenty Five"></ui5-tab>
+			<ui5-tab id="tabTwentySix" movable text="Twenty Six"></ui5-tab>
+			<ui5-tab id="tabTwentySeven" movable text="Twenty Seven"></ui5-tab>
+			<ui5-tab-separator></ui5-tab-separator>
+			<ui5-tab id="tabTwentyEight" movable text="Twenty Eight"></ui5-tab>
+			<ui5-tab id="tabTwentyNine" movable text="Twenty Nine"></ui5-tab>
+			<ui5-tab id="tabThirty" movable text="Thirty"></ui5-tab>
+		</ui5-tabcontainer>
+	</section>
+
+	<script>
+		const tc = document.getElementById("tabContainerDnd");
+		const maxNestingLevelInput = document.getElementById("maxNestingLevelInput");
+		let maxNestingLevel = 2;
+		maxNestingLevelInput.value = maxNestingLevel;
+
+		const getTabLevel = (element) => {
+			if (element.hasAttribute("ui5-tabcontainer")) {
+				return 0;
+			}
+
+			return 1 + getTabLevel(element.parentElement);
+		};
+
+		tc.addEventListener("ui5-move-over", (e) => {
+			const { source, destination } = e.detail;
+
+			if (!tc.contains(source.element)) {
+				return;
+			}
+
+			console.log(`Dragging "${source.element.id}" ${destination.placement.toLowerCase()} "${destination.element.id}"`);
+
+			if (destination.placement === "On" && (destination.element.id === "tabFour" || destination.element.id === "tabSeventeen")) {
+				return;
+			}
+
+			let potentialNestingLevel = getTabLevel(destination.element);
+
+			if (destination.placement === "On") {
+				potentialNestingLevel += 1;
+			}
+
+			if (potentialNestingLevel > maxNestingLevel + 1) {
+				return;
+			}
+
+			e.preventDefault();
+		});
+
+		tc.addEventListener("ui5-move", (e) => {
```

Comment:
the event "move" sounds a bit odd on its own. How about one of those:
move-done (+1 from me)
move-complete (+1 from Teo)
move-finish
move-ready
move-receive

Reply:
Hello,

I still believe just `move` is the best because it matches the rest of the events: `click`, `change` etc. and none of them are called `click-done` or `change-ready`.

One idea that comes to mind is to have an **event detail** in the future, if necessary, that describes the state of the event outcome, so the app can check: 

`if (e.detail.status === "failed") { // move operation failed` 

or something like this. Probably it would be easier for them to subscribe to just one event (`ui5-move`) and check the outcome in one event handler and do the respective work, rather than subscribing to `ui5-move-complete` and `ui5-move-failed` but I'm not 100% sure.

In the end, I don't have a strong opinion. I'd be fine with any of the options. both `move-done` and `move-complete` sound good to, if you decide to go with a two-word event name.

Reply:
Hi,
Ok, the discussion started from that we have drag-end and drop - which is a canceled drop and a successful drop, therefore we were thinking we need to mark the move as successful move.
But you are right - click, change are just single words.

Lets keep it just "move" then.

Thanks

Reply:
I will keep it "move" then.